functions/daemon.go

```diff
@@ -578,8 +591,19 @@ func UpdateKeys() error {
 }
 
 func holePunchWgPort(proto, portToStun int) (pubIP net.IP, pubPort int, natType string) {
-
-	pubIP, pubPort, natType = stun.HolePunch(portToStun, proto)
+	server := config.GetServer(config.CurrServer)
+	if server == nil {
+		server = &config.Server{}
+		server.Stun = true
+		stun.SetDefaultStunServers()
+	}
+	if server.Stun {
+		pubIP, pubPort, natType = stun.HolePunch(portToStun, proto)
```

Comment:
add this check inside stun.HolePunch func and just return if stun is false

Reply:
Done in new commit. please review again.